sdk/core/core/inc/az_http_request_builder.h

```diff
@@ -12,24 +12,38 @@
 #include <az_result.h>
 #include <az_span.h>
 
+#include <stdint.h>
+
 #include <_az_cfg_prefix.h>
 
 typedef struct {
   az_span buffer;
-  int16_t max_headers;
-  int16_t max_url_size;
   az_const_span method_verb;
-  int16_t retry_headers_start;
-  int16_t headers_end;
+  az_span url;
+  az_span body;
```

Comment:
This will probably change to a uStream in the future but it's OK for now.

Reply:
It was Victor's request to add body span. I am fine to change it in the future.

Reply:
I see this never really gets set anywhere so why was it added?
The customer has to pass the body to us at some point.
I say remove this this field for now.

Reply:
We will need it for KeyVault. I would keep the body for now until we have streaming.

Reply:
@vhvb1989, you will be using this, please tie-break :)

Reply:
yeah, we need a body to make a POST to get auth token
if it is not added here, then I would need to have an extra argument to `send_request` method to accept not only an http_builder but also a span to a body. witch would really make no sense "D

I will be very happy if you merge this as it is right now so I can make my PR pass and we can have a little more progress for our sprint goal.

Reply:
@antkmsft, I agree with Jeff.  Please wait to add it until its needed.  If Victor needs it he can add it with his PR

Reply:
Removed.